sklearn/partial_dependence.py

```diff
@@ -0,0 +1,648 @@
+        For regression and binary classification ``n_classes==1``.
+    """
+    n_samples = grid.shape[0]
+    X_eval = np.tile(X.mean(0), [n_samples, 1])
+    for i, variable in enumerate(target_variables):
+        X_eval[:, variable] = grid[:, i]
+    if est._estimator_type == 'regressor':
+        try:
+            pdp = est.predict(X_eval)
+        except NotFittedError:
+            raise ValueError('Call %s.fit before partial_dependence' %
+                             est.__class__.__name__)
+        if pdp.ndim != 1 and pdp.shape[1] == 1:
+            # Column output
+            pdp = pdp.ravel()
+        if pdp.ndim != 1 and pdp.shape[1] != 1:
+            # Multi-output
+            if not 0 <= output < pdp.shape[1]:
+                raise ValueError('Valid output must be specified for '
+                                 'multi-output models.')
+            pdp = pdp[:, output]
+        pdp = pdp[np.newaxis]
+    elif est._estimator_type == 'classifier':
+        try:
+            pdp = est.predict_proba(X_eval)
+        except NotFittedError:
+            raise ValueError('Call %s.fit before partial_dependence' %
+                             est.__class__.__name__)
+        if isinstance(pdp, list):
+            # Multi-output
+            if not 0 <= output < len(pdp):
+                raise ValueError('Valid output must be specified for '
+                                 'multi-output models.')
+            pdp = pdp[output]
+        pdp = np.log(np.clip(pdp, 1e-16, 1))
+        pdp = np.subtract(pdp, np.mean(pdp, 1)[:, np.newaxis])
+        pdp = pdp.transpose()
+    else:
+        raise ValueError('est must be a fitted regressor or classifier model.')
+    if pdp.shape[0] == 2:
+        # Binary classification
+        pdp = pdp[1, :][np.newaxis]
+    return pdp
+
+
+def partial_dependence(est, target_variables, grid=None, X=None, output=None,
+                       percentiles=(0.05, 0.95), grid_resolution=100,
+                       method=None):
+    """Partial dependence of ``target_variables``.
+
+    Partial dependence plots show the dependence between the joint values
+    of the ``target_variables`` and the function represented
+    by the ``est``.
+
+    Read more in the :ref:`User Guide <partial_dependence>`.
+
+    Parameters
+    ----------
+    est : BaseEstimator
+        A fitted classification or regression model.
+    target_variables : array-like, dtype=int
+        The target features for which the partial dependency should be
+        computed (size should be smaller than 3 for visual renderings).
+    grid : array-like, shape=(n_points, len(target_variables))
+        The grid of ``target_variables`` values for which the
+        partial dependency should be evaluated (either ``grid`` or ``X``
+        must be specified).
+    X : array-like, shape=(n_samples, n_features)
+        The data on which ``est`` was trained. It is used to generate
```

Comment:
As partial dependence plots are intended to show the response of the estimator at prediction time, I don't think so. Maybe I'm underthinking it. Anyone else care to chime in?